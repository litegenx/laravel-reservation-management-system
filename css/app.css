.router-link-active {
  cursor: pointer;
}

