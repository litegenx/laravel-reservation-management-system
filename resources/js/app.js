require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
