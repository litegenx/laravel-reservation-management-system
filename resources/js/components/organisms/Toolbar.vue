<template>
    <v-app-bar
        v-if="isActiveToolbar"
        app
        dark
        color="primary"
    >
        <v-app-bar-nav-icon
            v-if="isAuthenticated && isActiveDrawer"
            @click.stop="openDrawer"
        />
        <v-toolbar-title class="mr-5 align-center">
            <router-link
                to="/"
                tag="span"
            >
                <span class="title">{{ title }}</span>
            </router-link>
        </v-toolbar-title>
    </v-app-bar>
</template>

<script>
    import { mapGetters, mapActions } from 'vuex';

    export default {
        computed: {
            ...mapGetters({
                title: 'getTitle',
                isActiveToolbar: 'common/isActiveToolbar',
                isActiveDrawer: 'common/isActiveDrawer',
                isAuthenticated: 'auth/isAuthenticated',
            }),
        },
        methods: {
            ...mapActions({
                openDrawer: 'common/openDrawer',
            }),
        },
    };
</script>
