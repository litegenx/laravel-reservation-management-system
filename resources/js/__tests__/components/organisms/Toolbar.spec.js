import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import Toolbar from '../../../components/organisms/Toolbar';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

describe('Toolbar', () => {
    it('should not render toolbar if not active', () => {
        const wrapper = mount(app(Toolbar, 'top'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'common/isActiveToolbar': () => false,
                    'common/isActiveDrawer': () => false,
                    'auth/isAuthenticated': () => false,
                },
                actions: {
                    'common/openDrawer': () => {
                    },
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('not active');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-toolbar__content')).toHaveLength(0);
    });

    it('should not render toolbar toggle if not active drawer', () => {
        const wrapper = mount(app(Toolbar, 'top'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'common/isActiveToolbar': () => true,
                    'common/isActiveDrawer': () => false,
                    'auth/isAuthenticated': () => true,
                },
                actions: {
                    'common/openDrawer': () => {
                    },
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('drawer not active');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-toolbar__content')).toHaveLength(1);
        expect(wrapper.find('.title').text()).toBe('Reservation System');
        expect(wrapper.findAll('.v-app-bar__nav-icon')).toHaveLength(0);
    });

    it('should not render toolbar toggle if not authenticated', () => {
        const wrapper = mount(app(Toolbar, 'top'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'common/isActiveToolbar': () => true,
                    'common/isActiveDrawer': () => true,
                    'auth/isAuthenticated': () => false,
                },
                actions: {
                    'common/openDrawer': () => {
                    },
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('not authenticated');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-toolbar__content')).toHaveLength(1);
        expect(wrapper.find('.title').text()).toBe('Reservation System');
        expect(wrapper.findAll('.v-app-bar__nav-icon')).toHaveLength(0);
    });

    it('should render toolbar', () => {
        const openDrawer = jest.fn();
        const wrapper = mount(app(Toolbar, 'top'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'common/isActiveToolbar': () => true,
                    'common/isActiveDrawer': () => true,
                    'auth/isAuthenticated': () => true,
                },
                actions: {
                    'common/openDrawer': openDrawer,
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('active');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-toolbar__content')).toHaveLength(1);
        expect(wrapper.find('.title').text()).toBe('Reservation System');
        expect(wrapper.findAll('.v-app-bar__nav-icon')).toHaveLength(1);

        wrapper.find('.v-app-bar__nav-icon').trigger('click');
        expect(openDrawer).toBeCalled();
    });
});
