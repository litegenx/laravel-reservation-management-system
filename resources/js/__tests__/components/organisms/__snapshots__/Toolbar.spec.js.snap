// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should not render toolbar if not active: not active 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <!---->
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Toolbar should not render toolbar toggle if not active drawer: drawer not active 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <header
      class="v-sheet v-sheet--tile theme--dark v-toolbar v-app-bar v-app-bar--fixed primary"
      style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <!---->
         
        <div
          class="v-toolbar__title mr-5 align-center"
        >
          <span
            class="router-link-exact-active router-link-active"
          >
            <span
              class="title"
            >
              Reservation System
            </span>
          </span>
        </div>
      </div>
    </header>
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 64px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Toolbar should not render toolbar toggle if not authenticated: not authenticated 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <header
      class="v-sheet v-sheet--tile theme--dark v-toolbar v-app-bar v-app-bar--fixed primary"
      style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <!---->
         
        <div
          class="v-toolbar__title mr-5 align-center"
        >
          <span
            class="router-link-exact-active router-link-active"
          >
            <span
              class="title"
            >
              Reservation System
            </span>
          </span>
        </div>
      </div>
    </header>
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 64px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Toolbar should render toolbar: active 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <header
      class="v-sheet v-sheet--tile theme--dark v-toolbar v-app-bar v-app-bar--fixed primary"
      style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <button
          class="v-app-bar__nav-icon v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--dark"
            >
              $menu
            </i>
          </span>
        </button>
         
        <div
          class="v-toolbar__title mr-5 align-center"
        >
          <span
            class="router-link-exact-active router-link-active"
          >
            <span
              class="title"
            >
              Reservation System
            </span>
          </span>
        </div>
      </div>
    </header>
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 64px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;
